### PR TITLE
fix: don't double-panic in recovery middleware on empty error slices

### PR DESCRIPTION
**Pull request by fnoopv**
errors.NewSkip returns nil for nil or empty-slice panic values, so the unconditional type assertion panicked again inside the deferred recovery, breaking the connection instead of returning a 500. Use a comma-ok assertion and add a regression test for panic([]error{}).

## References

no

## Problem
`recoveryMiddleware` did an unconditional type assertion:

    e := errors.NewSkip(err, 4).(*errors.Error)

`errors.NewSkip` returns `nil` for a `nil` reason or an empty error
slice (`panic([]error{})`). In that case the assertion panics again
inside the deferred `recover`, so the connection is abruptly closed by
the `net/http` server instead of returning a `500`.

## Changes
Use a comma-ok type assertion so an unrepresentable panic value
(nil/empty slice) is handled gracefully: the status is still forced to
`500`, but no second panic occurs.

## Tests
Add `empty_slice_panic` regression test asserting the handler no longer
panics, returns `500`, and leaves no error/log behind.

### Possible drawbacks

no

### CODE REVIEW

File: middleware.go [hunk 1]
```diff
@@ -81,9 +81,10 @@ func (m *recoveryMiddleware) Handle(next Handler) Handler {
 		panicked := true
 		defer func() {
 			if err := recover(); err != nil || panicked {
-				e := errors.NewSkip(err, 4).(*errors.Error) // Skipped: runtime.Callers, NewSkip, this func, runtime.panic
-				m.Logger().Error(e)
-				response.err = e
+				if e, ok := errors.NewSkip(err, 4).(*errors.Error); ok { // Skipped: runtime.Callers, NewSkip, this func, runtime.panic
+					m.Logger().Error(e)
```
Comment (System-Glitch) [inline middleware.go]: Even if the error is `nil`, it should be printed in the logs.

File: middleware_test.go [hunk 2]
```diff
@@ -172,6 +172,31 @@ func TestRecoveryMiddleware(t *testing.T) {
...
+
+		request := NewRequest(httptest.NewRequest(http.MethodGet, "/test", nil))
+		response := NewResponse(server, request, httptest.NewRecorder())
+
+		require.NotPanics(t, func() {
+			handler(response, request)
+		})
+
+		assert.Nil(t, response.GetError())
+		assert.Empty(t, logBuffer.String())
```
Comment (System-Glitch) [inline middleware_test.go]: This test needs to be updated since we want to print the `nil` error.

**Review by System-Glitch — CHANGES_REQUESTED** [review]
Thanks a lot for finding this edge case!

**Comment by System-Glitch** [comment]
Thanks!
